Source/WebCore/platform/network/cocoa/ResourceResponseCocoa.mm

```diff
@@ -169,7 +169,12 @@ static inline AtomString extractHTTPStatusText(CFHTTPMessageRef messageRef)
 
         if (m_initLevel < CommonFieldsOnly) {
             m_url = [m_nsResponse URL];
-            m_mimeType = [m_nsResponse MIMEType];
+#if USE(AVIF)
+            if (m_url.string().endsWithIgnoringASCIICase(".avif"_s) || m_url.string().endsWithIgnoringASCIICase(".avifs"_s))
+                m_mimeType = "image/avif"_s;
+            else
+#endif
```

Comment:
Why are AVIFs special? I don't see any other sections for other formats.

Reply:
> Why are AVIFs special? I don't see any other sections for other formats.

Because AVIFs are special on down levels (i.e. USE(AVIF) is true). The system does not support them but WebKit does. So we should not rely on `[NSURLResponse MIMEType] `because it will give `application/octet-stream` for a URL to an AVIF image.